hadoop-common-project/hadoop-common/src/main/proto/RpcHeader.proto

```diff
@@ -157,6 +158,11 @@ message RpcResponseHeaderProto {
   optional bytes clientId = 7; // Globally unique client ID
   optional sint32 retryCount = 8 [default = -1];
   optional int64 stateId = 9; // The last written Global State ID
+  optional bytes routerFederatedState = 10; // Alignment context info for use with routers.
+}
+
+message RouterFederatedStateProto {
```

Comment:
I think this should not be here and should be an implementation detail in RBF.

Reply:
Moved to FederationProtocol.proto in the rbf module.